src/dbos.ts

```diff
@@ -1156,6 +1157,28 @@ export class DBOS {
     }
   }
 
+  /**
+   * Use dedupID to enqueue workflows started within the `callback`.
+   * @param options - enqueueOptions for the workflow enqueued
+   * @param callback - Function to run, which would call or start workflows
+   * @returns - Return value from `callback`
+   */
+
+  static async withEnqueueOptions<R>(options: EnqueueOptions, callback: () => Promise<R>): Promise<R> {
```

Comment:
I really think we should fold the `enqueueOptions` into `withWorkflowQueue`. given the poor dev experience of the `with***` methods, I really *don't* want the user to have to call two of them.